### PR TITLE
normalize phase error struct usage and and add path tracking to errors

### PR DESCRIPTION
**Pull request by benwilson512**
TODO:
- [x] Test indices in paths
- [x] test aliases in paths

### CODE REVIEW

**Comment by benwilson512** [comment]
@sikanhe

File: lib/absinthe/resolution.ex [hunk 1]
```diff
@@ -83,6 +83,45 @@ defmodule Absinthe.Resolution do
...
+
+  In this case `5` is the 0 based index in the list of users the field is currently
+  at.
+  """
+  def path(%{path: path}) do
+    path
+    |> Enum.reverse
+    |> Enum.drop(1)
+    |> Enum.map(&field_name/1)
+  end
```
Comment (benwilson512) [inline lib/absinthe/resolution.ex]: @sikanhe you may want to call this function from your tracing package.

File: lib/absinthe/phase/document/complexity/result.ex [hunk 2]
```diff
@@ -47,11 +47,11 @@ defmodule Absinthe.Phase.Document.Complexity.Result do
...
-    Phase.Error.new(
-      __MODULE__,
-      error_message(node, complexity, max),
-      location: location
-    )
+    %Phase.Error{
+      phase: __MODULE__,
+      message: error_message(node, complexity, max),
+      locations: [location],
+    }
```
Comment (benwilson512) [inline lib/absinthe/phase/document/complexity/result.ex]: @bruce note this small change. The `.new` function wasn't really doing anything anymore and it made it harder to enforce the correct keys.

**Comment by sikanhe** [comment]
Love it 💯

**Pull request merged**